packages/mds-repository/connection-manager.ts

```diff
@@ -42,15 +46,16 @@ const {
   PG_MIGRATIONS = 'true' // Enable migrations by default
 } = process.env
 
-const connectionName = (prefix: string, mode: ConnectionMode) => `${prefix}-${mode}`
-
 export type ConnectionManagerOptions = Partial<PostgresConnectionOptions>
 
 export const ConnectionManager = (prefix: string, options: Omit<ConnectionManagerOptions, 'cli'> = {}) => {
-  let connections: Connection[] | null = null
+  const lock = new AwaitLock()
```

Comment:
Mutex + Node still doesn't entirely compute in my brain, but I understand in essence how it's working with async patterns. I wish node was multithreaded *sigh*

Reply:
Agreed, but this does resolve the issue quite nicely.